client/src/get-ticket/components/home.tsx

```diff
@@ -1,14 +1,45 @@
-import { useState } from 'react'
+import { useEffect } from 'react'
 import Button from 'react-bootstrap/Button';
 import '../get-ticket.css';
 import { useNavigate } from 'react-router-dom';
+import API from '../../API/API.tsx';
 
-function Home() {
+function Home(props: any) {
 
+    const dirty = props.dirty;
+    const setDirty = props.setDirty;
+
+    useEffect(() => {
+        setDirty(true);
+      }, [])
+
+    // get all the services
+    useEffect(() => {
+        if (dirty) {
+          API.getAllServices()
+            .then(services => {
+              props.setServices(services);
+              setDirty(false);
+            });
+        }
+    }, [dirty]);
+
+    const AllServices = props.services;
```

Comment:
This new definition is quite redundant. I believe "services" already indicates the cause.